File: ios/CoChat/Repositories/UsersRepository.swift
import Foundation

final class UsersRepository {
    private let api: APIClient
    private let tokenStore: TokenStore

    init(api: APIClient, tokenStore: TokenStore) {
        self.api = api
        self.tokenStore = tokenStore
    }

    func me() async throws -> User {
        let res: UserEnvelope = try await api.get("users/me")
        return res.user
    }

    func updateProfile(fullName: String?, designation: String?) async throws -> User {
        let res: UserEnvelope = try await api.patch("users/me", body: UpdateProfileRequest(fullName: fullName, designation: designation))
        tokenStore.updateUser(res.user)
        return res.user
    }

    func uploadAvatar(data: Data, fileName: String, mimeType: String) async throws -> User {
        let res: UserEnvelope = try await api.postMultipart(
            "users/me/avatar",
            fileField: "avatar", fileData: data, fileName: fileName, mimeType: mimeType
        )
        tokenStore.updateUser(res.user)
        return res.user
    }

    func list(query: String = "") async throws -> [User] {
        let res: UsersEnvelope = try await api.get("users", query: ["q": query])
        return res.users
    }

    func get(id: String) async throws -> User {
        let res: UserEnvelope = try await api.get("users/\(id)")
        return res.user
    }
}

File: android/app/src/main/java/com/cochat/app/data/repository/UsersRepository.kt
package com.cochat.app.data.repository

import com.cochat.app.data.model.User
import com.cochat.app.data.remote.ApiService
import com.cochat.app.data.remote.TokenStore
import com.cochat.app.data.remote.UpdateProfileRequest
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.MultipartBody
import okhttp3.RequestBody.Companion.asRequestBody
import java.io.File

class UsersRepository(private val api: ApiService, private val tokenStore: TokenStore) {

    suspend fun me(): User = api.me().user

    suspend fun updateProfile(fullName: String?, designation: String?): User {
        val user = api.updateMe(UpdateProfileRequest(fullName, designation)).user
        tokenStore.updateUser(user)
        return user
    }

    suspend fun uploadAvatar(file: File, mimeType: String): User {
        val body = file.asRequestBody(mimeType.toMediaType())
        val part = MultipartBody.Part.createFormData("avatar", file.name, body)
        val user = api.uploadAvatar(part).user
        tokenStore.updateUser(user)
        return user
    }

    suspend fun list(query: String = ""): List<User> = api.listUsers(query).users

    suspend fun get(id: String): User = api.getUser(id).user
}
